gc/2022/GC-08-23.md

```diff
@@ -33,3 +33,187 @@ Installation is required, see the calendar invite.
+
+BT: I think we would call this layer 1 compression, I see what we’re doing  as we’re trying to use the type reconstruction algorithm as a way to compression, if we really want to use this, that might be really cool. That’s a compression mechanism that might not be available to a general purpose zip algorithm, if we were to continue looking at that idea, that could exist at another level, and we can have another tool that decompresses, still get the best of both worlds
+
+AZ: Fair point that the compression could be done at another level, we haven’t focused on micro optimizing things. Want to push back against some of these, while in theory we could do something in the space, but it would take time, it’s unclear that we would actually do this. Unclear that we would actually have the resources to work on this, there are things we could have done the past, but did not but those are things  (const 0, const 1) that compress well, on the other hand, type annotations are high entropy thing and have a higher cost, this stuff does matter, we have a lot of use cases on teh web where there are limited on the size of the download, while it doesn’t matter in some countries with unlimited bandwidth, other countries it isn’t. 5% is significant, lots of efforts to reduce code size overall and ever % matters. IT is likely that if we don’t save this now, it’s possible that we’ll be stuck with this for a very long time and that would be wasteful.
+
+DG: We’ve had discussions on both sides of this, one of the things that I’d like to circle back on, was that there are future features  that there is a point of time in the future that we might feel comfortable about not having type annotations, you mentioned type parameters, maybe we can dig into that a little more what are other examples of future features? It would be nice to quantify that, at what point of time do we think the risk is lower, or is there a feature set you have in mind?
+
+AR: It’s really a spectrum, certainty is increasing monotonically, at some point we can have a more informed decision about this. What the features would be, or what the timeline would be, I would make no predictions at this point given how long we’ve worked on at this point
+
+DG: phrased this wrong, maybe looking for more of an opinion, the point that this could be unbounded w.r.t to time/features it seems uncomfortable at a time that we see the binary size being bloated that we want to make sure that we’re not adding size increase in unless necessary
+
+FM: What would be the rationale for having type parameters? I’m not sure that there is one
+
+AR: Currently there is no way to create code that operates the same on any reference, where you don’t really care  what the reference is. This is a frequent thing for compilers, in Wasm you can’t express them because every reference has to be concretely defined, i view  this feature as giving a way to generate this kind of code that native code compilers are able to generate
```

Comment:
```suggestion
AR: Currently there is no way to create code that operates the same on any reference, where you don’t really care  what the reference is. This is a frequent thing for compilers, in Wasm you can’t express them because every reference has to be concretely defined, I view this feature as giving a way to generate this kind of code that native code compilers are able to generate
```